packages/code/src/main.rs

```diff
@@ -0,0 +1,144 @@
+                    {"name": "theme", "default": "mocha", "description": "Theme of the code section"},
+                    {"name": "bg", "default": "default", "description": "Background of the code section"},
+                ],
+            }
+        ]
+        }
+    ))
+    .unwrap());
+}
+
+fn transform(from: &String, to: &String) {
+    match from.as_str() {
+        "code" => transform_code(to),
+        other => {
+            eprintln!("Package does not support {other}");
+            return;
+        }
+    }
+}
+
+fn transform_code(to: &String) {
+    macro_rules! get_arg {
+        ($input:expr, $arg:expr) => {
+            if let Value::String(val) = &$input["arguments"][$arg] {
+                val
+            } else {
+                panic!("No theme argument was provided");
+            }
+        }
+    }
+    match to.as_str() {
+        "html" => {
+            let input: Value = {
+                let mut buffer = String::new();
+                io::stdin().read_to_string(&mut buffer).unwrap();
+                serde_json::from_str(&buffer).unwrap()
+            };
+
+            let code = input["data"].as_str().unwrap();
+            let lang = get_arg!(input, "lang");
+            let font_size = get_arg!(input, "font_size");
+            let tab_size = get_arg!(input, "tab_size");
+            let theme = get_arg!(input, "theme");
+            let bg = get_arg!(input, "bg");
+
+
+            let (highlighted, default_bg) = get_highlighted(code, lang, theme);
+            let style = get_style(font_size, tab_size, bg, default_bg);
+
+            let html = format!(r#"<pre {style}>{highlighted}</pre>"#);
+            let json = json!({"name": "raw", "data": html}).to_string();
+
+            print!("[{json}]");
+        }
+        other => {
+            eprintln!("Cannot convert code to {other}");
+            return;
+        }
+    }
+}
+
+fn get_style(font_size: &String, tab_size: &String, bg: &String, default_bg: Option<Color>) -> String {
+    let hex;
+    if bg == "default" && default_bg.is_some() {
+        let c = default_bg.unwrap();
+        hex = format!("{:02x}{:02x}{:02x}", c.r, c.g, c.b);
+    } else {
+        hex = bg.clone();
+    }
+
+    let mut style = String::from("style=\"");
+    write!(style, "box_sizing: border_box; ").unwrap();
+    write!(style, "padding: 0.5rem; ").unwrap();
+    write!(style, "tab-size: {}; ", tab_size).unwrap();
+    write!(style, "font-size: {}px; ", font_size).unwrap();
+    write!(style, "background-color: #{}; ", hex).unwrap();
+    style.push('\"');
+    style
+}
+
+fn get_highlighted(code: &str, lang: &String, tm: &String) -> (String, Option<Color>) {
+    let ss = SyntaxSet::load_defaults_newlines();
+    let ts = ThemeSet::load_defaults();
+    let syntax = match ss.find_syntax_by_token(lang) {
+        Some(sr) => sr,
+        _ => ss.find_syntax_by_token("py").unwrap()
```

Comment:
This is kind of strange, it seems like if tries to find the language from the user input but if it doesn't find it, it defaults to python. This means that if you do `[code cpp]` and `cpp` isn't found (if it is called `c++` instead), `python` formatting will be used. I don't think this is good; if the user specifically asks for a language syntax highligter and it doesn't exist, it should error. Maybe you could add `py` as the default argument instead?

Reply:
Also, it would be nice to know what languages are available, or how to find a list of them or something of the like

Reply:
Fair point. The reason I did this was to avoid repeated panics as the user is typing out the argument. If this isn't something we should care about, I can definitely change it. I agree that silently defaulting to python like this can cause unwanted behaviour though.

As for available languages, I haven't found a list yet. I know it uses Sublime Text's default syntax definitions so maybe I can find something there. If I do find something, where would I put this information?

Reply:
Is there maybe a good way to use your system for warnings for this?

Reply:
You could also do warnings, it is your choice, and I guess that you'll have to default to outputting text without syntax highlighting then. Warnings sound good as well! You do them by printing your warning text using `eprintln!(...)` and then continuing as normal, printing your JSON output to `stdout`